src/main/java/fr/sncf/osrd/railjson/parser/RailJSONParser.java

```diff
@@ -188,11 +188,45 @@ else if (rjsSignal.applicableDirection == ApplicableDirection.REVERSE)
             signalsBuilder.build();
         }
 
-        // Fill switch with their right / left track sections
+        // Fill switch ports (ie connected track section endpoints)
         for (var rjsSwitch : railJSON.switches) {
             var switchRef = switchNames.get(rjsSwitch.id);
-            switchRef.leftTrackSection = infraTrackSections.get(rjsSwitch.left.section.id);
-            switchRef.rightTrackSection = infraTrackSections.get(rjsSwitch.right.section.id);
+            for (var entry : rjsSwitch.ports.entrySet()) {
```

Comment:
In this part of the code try to use as less as possible `getKey()` and `getValue()` to do so use local variables with relevant names.